File: 4055-longest-balanced-substring-i/longest-balanced-substring-i.cpp
class Solution {
public:
    int longestBalanced(string s) {
        int n = s.size();
        int ans = 0;

        for (int i = 0; i < n; i++) {
            
            unordered_map<char, int> freq;
            for (int j = i; j < n; j++) {
                freq[s[j]]++;

                int val = 0;
                bool ok = true;
                for (auto &p : freq) {
                    if (val == 0) val = p.second;
                    else if (p.second != val) {
                        ok = false;
                        break;
                    }
                }

                if (ok) ans = max(ans, j - i + 1);
            }
        }

        return ans;
    }
};

File: 4055-longest-balanced-substring-i/longest-balanced-substring-i.py
class Solution:
    def longestBalanced(self, s: str) -> int:
        ans = 0
        n = len(s)

        i = 0
        while i < n:
            mp = {}
            j = i

            while j < n:
                mp[s[j]] = mp.get(s[j], 0) + 1

                ok = True
                prev = mp[s[j]]

                for key, val in mp.items():
                    if val != prev:
                        ok = False
                        break

                if ok:
                    ans = max(ans, j - i + 1)

                j += 1

            i += 1

        return ans
